vendor/github.com/status-im/ottoext/fetch/fetch.go

```diff
@@ -12,6 +12,7 @@ import (
 
 	"github.com/status-im/ottoext/loop"
 	"github.com/status-im/ottoext/promise"
+	"github.com/status-im/status-go/geth/jail/vm"
```

Comment:
I'm definitely against such dependencies.
Can we do it through an interface defined locally in `ottoext` to decouple this dependency. Maybe, it's worth for `JailCell` to embed this new interface into itself.

Reply:
Do you mean moving VM wrapper to `ottoext` and defining it as an interface? We've discussed this above with @adambabik and it looks like possible solution, but if we going to introduce new packages to `ottoext` we'd better  own `ottoext` package completely and make it part of `status-go`.

Reply:
I've thought we're already owning it? In a sense that we're not going to sync with the upstream.
No, I mean creating an interface like `JailCell` in `ottoext` here and depending upon it everywhere in `ottoext`.

Reply:
Okay, I think this is a moment when it's better to take a decision and choose between:
 - we keep `ottoext` as an external dependency, trying to change it as little as possible
 - make it part of `status-go` as we're, in fact, hacking it substancially

If we go with second, I would move into the `status-go` directory, and discussed problem will be much easier solve and to reason about.

Reply:
This is already a substantially modified version so I think we should for the second option.
But it doesn't eliminate the need to decouple the dependency: `JailCell` is our domain concept while otto vm is a means to handle vm. The latter shouldn't important the first, in my opinion.

Reply:
Will you do it under this PR?

Reply:
Since this PR relates with fetch but in truth, heavily relies on the changes on ottoext, it should be done here. I see no reason, having a second PR for vendor updates. 

@divan When you say `make it part of status-go`, do you mean bring in as a subpackage of `status-go`, does the use of the fork on the `status-im` repo not meet the requirement of it being part of our libs already?

Reply:
Can we just make `vm` package part of `ottoext` and bring that into `jail`. i find the reverse dependency chain not good. We could use an interface like @tiabc suggested.

Reply:
@influx6 exactly. The thing is `ottoext` is kinda "forked depencency" but we're going to modify it to the point where it's impossible to apply changes in original lib (if any). Even more, original design is incompatible with our goals (long time loop) and have some strange design decisions.

That said I see no reason keep this 2 levels of indirect fork (upstream -> fork -> vendored fork), knowing that is permanently out of sync with upstream, and, in fact, already a different lib that we going to hack even more.

I strongly feel that the best solution is to move into `geth/jail` package. Let's say we'll have `geth/jail/loop`, `geth/jail/fetch` and `geth/jail/timers`.

Reply:
Alright, sounds good. @divan.

@tiabc I think we should do this in this commit.